### PR TITLE
Empty Where Filter Input on Count Endpoints

### LINKED ISSUE #309: Empty Dict on Query Param for Count Endpoints
**Description**:
On count endpoints, the following query param doesn't work (400 is returned):
`where={}`

However, on non-count endpoints, the following query param does work and basically behaves in a way that there are no filters:
`filter={}`

We should make the count version have a similar behaviour as the non-count version.

I only noticed this while writing tests with `pytest.param()`, trying to reduce duplication. I don't know how much a user would use an empty dict but perhaps they might. This might be a case of fixing if we have time, but not stressing about it if we can't. If so, the tests that I've skipped can be easily rewritten in a number of minutes to avoid this bug.

**Acceptance criteria**:
- [ ] Make `where={}` work on count endpoints
- [ ] Unskip relevant tests in `test_count_dataset_files.py` and `test_count_endpoint.py`

### PR DESCRIPTION
**Pull request by VKTB**
This PR will close #309 

## Description
The changes in this PR make the count endpoints work when they receive an empty where filter input (`where={}`). Also added some extra code so that `FilterError` is raised when the where filter input is not an object.

## Testing Instructions
- [ ] Review code
- [ ] Check GitHub Actions build
- [ ] If `icatdb Generator Script Consistency Test` CI job fails, is this because of a deliberate change made to the script to change generated data (which isn't actually a problem) or is here an underlying issue with the changes made?
- [ ] Review changes to test coverage
- [ ] Does this change mean a new patch, minor or major version should be made? If so, does one of the commit messages feature `fix:`, `feat:` or `BREAKING CHANGE:` so a release is automatically made via GitHub Actions upon merge?

### CODE REVIEW

**Comment by VKTB** [comment]
The failing test jobs will be fixed by #325

File: datagateway_api/src/search_api/query_filter_factory.py [hunk 1]
```diff
@@ -103,87 +103,98 @@ def get_where_filter(where_filter_input, entity_name):
...
+                where_filters.append(nested)
+            elif list(where_filter_input.keys())[0] == "text":
+                log.debug("Text operator found within JSON where object")
+                try:
+                    entity_class = getattr(search_api_models, entity_name)
+                except AttributeError as e:
+                    raise SearchAPIError(
+                        f"No text operator fields have been defined for {entity_name}"
+                        f", {e.args}",
+                    )
```
[the remark this answers is not in the document]
Comment (MRichards99) [reply datagateway_api/src/search_api/query_filter_factory.py:153]: Agreed, that message doesn't fit the code wrapped around it. We should probably rewrite that message because it still could happen (e.g. if you call `get_where_filter()` with `BananaEntity` as the `entity_name`). Something along the lines of the following would fit better (feel free to reword):

```python
f"No model for {entity_name} could be found, a different entity name should be used."
```

As for line 156, I guess technically an exception _could_ be raised, but all the entity classes in `models.py` have `_text_operator_fields` defined in them, even if that entity has no text operator fields (an empty list is stored in that case).  In that situation, line 160 (and the logging in that conditional) covers that so I don't think any additional code changes are needed for that section?

In summary, I think a re-wording of that error message will suffice

**Review by MRichards99 — CHANGES_REQUESTED** [review]
Looks good, just need to reword the error message as per my comment and merge master in once #325 is merged so the CI passes.

File: datagateway_api/src/search_api/query_filter_factory.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (VKTB) [reply datagateway_api/src/search_api/query_filter_factory.py]: Fixed in https://github.com/ral-facilities/datagateway-api/pull/326/commits/263ad34be42fe57e1e7ebf67e0a534d505b78a2a

**Review by MRichards99 — DISMISSED** [review]
Looks good, we should probably wait to merge this PR so we can keep one in reserve for the DOI minting discussed on Slack.

**Comment by MRichards99** [comment]
This was waiting for a DOI to be minted for the search API. I believe this is unlikely to happen now, so the merge conflicts should be resolved and the PR should be merged.
